dre-service/src/main/java/io/cdap/re/RulesDB.java

```diff
@@ -630,4 +636,13 @@ private long getCurrentTime() {
 
     return keyFields;
   }
+
+  private List<String> getRulebookRulesFromTable(StructuredRow rulebookRow) {
+    String rulesColumn = Objects.toString(rulebookRow.getString(RULES_COL), "");
+    List<String> rules = new ArrayList<>(Arrays.asList(rulesColumn.split(",")));
+    rules.removeIf(item -> item == null || "".equals(item));
```

Comment:
It doesn't seem right that empty rules are written in the first place. They should be removed immediately where they are added, instead of stored and then filtered out on reading.

Reply:
Empty or null rules are not written to the StucturedTable. The problem is that `rulebookRow.getString(RULES_COL)` returns null if there are no rules. Hence, why I call `Objects.toString(...)` passing a default value of an empty string. Then, when calling `List<String> rules = new ArrayList<>(Arrays.asList(rulesColumn.split(",")));` on an empty string, it create a list with one element (i.e., the empty string). That is why call `rules.removeIf(...)` to remove that empty string.

I realize that removing empty or null elements in the list is very specific. Is there a better way to handle this case?

Reply:
I see. It is clearer if you do a null check on the result of rulebookRow.getString(RULES_COL). If it's null, return an empty list. Otherwise return Arrays.asList(...)

Reply:
Done